openBCIBoard.js

```diff
@@ -1,14 +1,471 @@
+        }
+    };
+
+    OpenBCIBoard.prototype.sampleRate = function() {
+        if(this.daisy) {
+            return k.OBCISampleRate125;
+        } else {
+            return k.OBCISampleRate250;
+        }
+    };
+
+    OpenBCIBoard.prototype.numberOfChannels = function() {
+        if(this.daisy) {
+            return k.OBCINumberOfChannelsDaisy;
+        } else {
+            return k.OBCINumberOfChannelsDefault;
+        }
+    };
+
+    OpenBCIBoard.prototype.processBytes = function(data) {
+        var self = this;
+        var sizeOfData = data.byteLength;
+        if(self.lookingForMoney) { //in a reset state
+            for (var i = 0; i < sizeOfData - 2; i++) {
+                if (self.moneyBuf.equals(data.slice(i, i + 3))) {
+                    console.log('Money!');
+                    self.lookingForMoney = false;
+                    self.emit('ready');
+                }
+            }
+        } else { //ready to open the serial fire hose
+            // send input data to master buffer
+            self.bufMerger(self,data);
+
+            // parse the master buffer
+            while(self.masterBuffer.packetsRead < self.masterBuffer.packetsIn) {
+                var rawPacket = self.bufPacketStripper(self);
+                var newSample = OpenBCISample.convertPacketToSample(rawPacket);
+                if(newSample) {
+                    //console.log('Wow, for the first time, you actually got a packet... maybe lets check it out!');
+                    self.emit('sample', sample);
+                } else {
+                    self.badPackets++;
+                    /*TODO: initiate messed up packet protocol, need to sync back up with good packet */
+                }
+                OpenBCISample.convertPacketToSample(rawPacket).then(function(sample) {
+
+                }, function(err) {
+
+                });
+
+            }
+        }
+    };
+
+    OpenBCIBoard.prototype.autoFindOpenBCIBoard = function(callback) {
+        var macSerialPrefix = 'usbserial-D';
+        var self = this;
+        serialPort.list(function(err, ports) {
+            console.log('Searching for ports...');
+            var foundPort = false;
+            ports.forEach(function (port) {
+                if (port.comName.search(macSerialPrefix) > 0) {
+                    self.portName = port.comName;
+                    console.log('found');
+                    callback(port.comName);
+                    foundPort = true;
+                }
+            });
+            if(!foundPort) {
+                callback(null,ports);
+            }
+            //return false;
+            //reject('unable to auto locate bci device');
+        })
+    };
+
+    OpenBCIBoard.prototype.debugSession = function() {
+        if(this.badPackets > 0) {
+            console.log('Dropped a total of ' + this.badPackets + ' packets.');
+        }
+    };
+
+    OpenBCIBoard.prototype.bufMerger = function(self,inputBuffer) {
```

Comment:
I added this as a comment... does this help a little?

"loose bytes results when there is not an even multiple of packets in the inputBuffer
example: The first time this is ran there are only 68 bytes in the first call to this function therefore there are only two packets (66 bytes), these extra two bytes need to be save for the next call and be considered in the next iteration so we can keep track of how many bytes we need to read."